reconcile/change_owners/change_owners.py

```diff
@@ -231,6 +232,31 @@ def init_gitlab(gitlab_project_id: str) -> GitLabApi:
     return GitLabApi(instance, project_id=gitlab_project_id, settings=settings)
 
 
+def assert_restrictive(
+    changes: list[BundleFileChange], user: str, approval_comments: list[dict[str, Any]]
+) -> None:
+    for change in changes:
+        for dc in change.diff_coverage:
+            for c in dc.coverage:
+                if c.change_type_processor.restrictive:
+                    approvers = {a.org_username for a in c.approvers}
+                    good_to_test_approvers = {a["username"] for a in approval_comments}
+
+                    for gttapprover in good_to_test_approvers:
+                        if gttapprover in approvers:
+                            logging.info(
+                                f"Restrictive change approved by {gttapprover}"
+                            )
+                            return
+
+                    if user not in approvers:
+                        logging.error(
+                            f"change type {c.change_type_processor.name} is restrictive"
+                            f"user  {user} is not an approver"
+                        )
+                        sys.exit(1)
```

Comment:
i think it would be better to signal failed assertion back to the caller and handle the exit there.
do we want to leave a message on the MR to help the author to figure out what is going on or is the output in the PR check logs enough?

at least we would need to mention who are the approvers and who would need to give their `/good-to-test`. can we add that somehow to the existing MR comment with all the approvers listed?

Reply:
also, for MRs where AppSRE is the approriate approver i would love that the MR to shows up in the review queue

Reply:
Great feedback. I'll try to make it easier to digest for the user